src/site/content/en/blog/javascript-this/index.md

```diff
@@ -0,0 +1,356 @@
+
+```js
+function someFunction() {
+  return this;
+}
+
+const boundObject = {hello: 'world'};
+const boundFunction = someFunction.bind(boundObject);
+```
+
+Whenever `boundFunction` is called, its `this` value will be the object passed to `bind`, `{foo: 'bar'}`.
+
+```js
+// Logs `false`:
+console.log(someFunction() === boundObject);
+// Logs `true`:
+console.log(boundFunction() === boundObject);
+```
+
+{% Aside 'warning' %}
+Avoid using `bind` to bind a function to its outer `this`. Instead, use [arrow functions](#arrow-functions), as they make `this` clear from the function declaration, rather than something that happens later in the code.
+
+Don't use `bind` to set `this` to some value unrelated to the parent object; it's usually unexpected and it's why `this` gets such a bad reputation. Consider passing the value as an argument instead; it's more explicit, and works with arrow functions.
+{% endAside %}
+
+### Other examples
+
+When calling a bound function, the value of `this` _can't_ be changed with [`call` or `apply`](#call-apply):
+
+```js
+// Logs `true` - called `this` value is ignored:
+console.log(boundFunction().call({foo: 'bar'}) === boundObject);
+// Logs `true` - applied `this` value is ignored:
+console.log(boundFunction().apply({foo: 'bar'}) === boundObject);
+```
+
+When calling a bound function, the value of `this` _can't_ be changed by calling the function as a member of another object:
+
+```js
+const obj = {boundFunction};
+// Logs `true` - parent object is ignored:
+console.log(obj.boundFunction() === boundObject);
+```
+
+## Otherwise, if `this` is set all call-time: {: #call-apply }
+
+```js
+function someFunction() {
+  return this;
+}
+
+const someObject = {hello: 'world'};
+
+// Logs `true`:
+console.log(someFunction.call(someObject) === someObject);
+// Logs `true`:
+console.log(someFunction.apply(someObject) === someObject);
+```
+
+The value of `this` is the object passed to `call`/`apply`.
+
+{% Aside 'warning' %}
+Don't use `call`/`apply` to set `this` to some value unrelated to the parent object; it's usually unexpected and it's why `this` gets such a bad reputation. Consider passing the value as an argument instead; it's more explicit, and works with arrow functions.
+{% endAside %}
+
+Unfortunately `this` is set to some other value by things like DOM event listeners, and using it can result in difficult-to-understand code:
+
+{% Compare 'worse' %}
+
+```js
+element.addEventListener('click', function (event) {
+  // Logs `element`, since the DOM spec sets `this` to
+  // the element the handler is attached to.
+  console.log(this);
+});
+```
+
+{% endCompare %}
+
+I avoid using `this` in cases like above, and instead:
+
+{% Compare 'better' %}
+
+```js
+element.addEventListener('click', (event) => {
+  // Grab it from the parent scope:
+  console.log(element);
+  // Or from the event object:
+  console.log(event.currentTarget);
+});
+```
+
+{% endCompare %}
+
```

Comment:
How about adding a quick sentence listing some more common examples of built-in Web Platform APIs that force a specific `this` value (e.g. `setTimeout` & `setInterval`, whose forced `this` is IMHO even more surprising than event handlers)?

Reply:
I've never really been surprised by `this` in `setTimeout`. It's never a useful value so it doesn't really get used. Do you have an example in mind?

Reply:
> It's never a useful value so it doesn't really get used.

This is exactly why it's surprising, IMHO. At least for event listeners, I can see the point of giving easy access to the affected node. I have no idea why you'd need a shorthand for `globalThis` within `setTimeout`/`setInterval`, which makes the API design more baffling.